wagtail/snippets/views/snippets.py

```diff
@@ -162,6 +162,12 @@ class IndexView(generic.IndexViewOptionalFeaturesMixin, generic.IndexView):
     page_kwarg = "p"
     table_class = InlineActionsTable
 
+    def get_add_url(self):
+        url = reverse(self.add_url_name)
+        if self.locale:
+            url += "?locale=" + self.locale.language_code
+        return url
```

Comment:
If this is rebased on top of #11232, it should be pretty easy to follow the pattern in that PR using `_set_locale_query_param`. That PR hasn't been reviewed though, so it might change